internals/src/error.rs

```diff
@@ -26,3 +26,31 @@ macro_rules! write_err {
         }
     }
 }
+
+/// This module is used internally to check that error types implement certain traits.
+#[cfg(feature = "trait-checks")]
```

Comment:
Maybe we should use bare `cfg`, not feature, so this doesn't get pulled in when `--all-features` is used?

Reply:
I used `#[cfg(trait_checks)]` and I enabled it in all builds by exporting `RUSFLAGS` from the top level `contrib/test.sh` script.

Reply:
We shoud have at least one build without it in case something gets messed up.

Reply:
Oh yes, very good point.